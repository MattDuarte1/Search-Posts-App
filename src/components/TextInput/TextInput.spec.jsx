import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should have a value of inputValue', () => {
    const fn = jest.fn();
    render(<TextInput inputValue={'testando'} handleChange={fn} />);
    const input = screen.getByPlaceholderText(/pesquise aqui/i);
    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} inputValue="o valor" />);

    const input = screen.getByPlaceholderText(/pesquise aqui/i);
    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(
      <TextInput inputValue="testando" handleChange={fn} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
