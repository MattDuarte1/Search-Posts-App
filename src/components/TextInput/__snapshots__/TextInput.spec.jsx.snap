// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="pesquise aqui"
  type="search"
  value="testando"
/>
`;
